starter-code/src/viking.js

```diff
@@ -1,11 +1,79 @@
 // Soldier
-function Soldier () {}
-
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+Soldier.prototype.attack = function() {
+  return this.strength;
+};
+Soldier.prototype.receiveDamage = function(damage) {
+  this.health = this.health - damage;
+};
 // Viking
-function Viking () {}
-
+function Viking(name, health, strength) {
+  this.name = name;
+  // this.health = health;
+  // this.strength = strength;
+  Soldier.call(this, health, strength);
+}
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.receiveDamage = function(damage) {
+  this.health = this.health - damage;
+  return (this.health>0) ? this.name + " has received " + damage + " points of damage" : this.name + " has died in act of combat";
+};
+Viking.prototype.battleCry = function() {
+  return "Odin Owns You All!";
+};
 // Saxon
-function Saxon () {}
-
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+}
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.receiveDamage = function(damage) {
+  this.health = this.health - damage;
+  return (this.health>0) ? "A Saxon has received " + damage + " points of damage" : "A Saxon has died in combat";
+};
 // War
-function War () {}
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+}
+War.prototype.addViking = function(viking) {
+  if (viking instanceof Viking) {
+    this.vikingArmy.push(viking);
+  }
+};
+War.prototype.addSaxon = function(saxon) {
+  if (saxon instanceof Saxon) {
+    this.saxonArmy.push(saxon);
+  }
+};
+War.prototype.vikingAttack = function() {
+  saxonRandom = Math.floor(Math.random() * this.saxonArmy.length);
+  vikingRandom = Math.floor(Math.random() * this.vikingArmy.length);
+  this.saxonArmy[saxonRandom].receiveDamage(this.vikingArmy[vikingRandom].strength);
```

Comment:
Aquí te recomiendo que this.saxonArmy[saxonRandom] lo metas en una variable, porque luego lo vuelves a usar. Y lo mismo con this.vikingArmy[vikingRandom].strength, así quedaría más legible esa línea.